prod/native/building/cmake/elastic_set_default_build_options.cmake

```diff
@@ -15,7 +15,7 @@ set(CMAKE_DISABLE_IN_SOURCE_BUILD ON)   # https://github.com/ComputationalRadiat
 set(CMAKE_DISABLE_SOURCE_CHANGES  ON)
 set(CMAKE_CXX_EXTENSIONS OFF)           # https://cmake.org/cmake/help/latest/prop_tgt/CXX_EXTENSIONS.html#prop_tgt:CXX_EXTENSIONS
 set(CMAKE_CXX_STANDARD_REQUIRED ON)     # https://cmake.org/cmake/help/latest/prop_tgt/CXX_STANDARD_REQUIRED.html#prop_tgt:CXX_STANDARD_REQUIRED
-set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_CXX_STANDARD 23)
```

Comment:
bumped cpp supported standard to c++23 (just to use new string_view constructor from ranges)